CMakeLists.txt

```diff
@@ -8,9 +8,18 @@ cmake_minimum_required(VERSION 3.1) # for "CMAKE_CXX_STANDARD" version
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake") # custom CMake modules like FindSQLiteCpp
 project(SQLiteCpp VERSION 2.99)
 
-# SQLiteC++ 3.x now requires C++11 compiler
-set(CMAKE_CXX_STANDARD 11)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
+# SQLiteC++ 3.x can make use of C++11 features
+if (NOT CMAKE_CXX_STANDARD)
+    set(CMAKE_CXX_STANDARD 11)
+endif ()
+# googletest requires C++11 features
+if (SQLITECPP_BUILD_TESTS)
+    if (CMAKE_CXX_STANDARD LESS 11)
+        message(STATUS "googletest requires c++ standard 11 or higher")
+    endif ()
+    message(STATUS "Setting c++ standard ${CMAKE_CXX_STANDARD} to required for unit tests with googletest")
+    set(CMAKE_CXX_STANDARD_REQUIRED ON)
```

Comment:
I don't really see the point of (CMAKE_CXX_STANDARD_REQUIRED ON) since my understanding is that it will enforce C++11 usage even if the user has selected C++17

=> isn't it possible to use a fatal error instead of the above message

actually, I am not sure of the behavior of the CMAKE_CXX_STANDARD variable if we try to set it to a higher version than supported?

Reply:
The point of `CMAKE_CXX_STANDARD_REQUIRED ON` was that I assumed you wanted to still support older compilers, and without this setting `c++11` was recommended but not required.

But no worries about newer standards. When setting `CMAKE_CXX_STANDARD_REQUIRED ON` it just means "enforce whatever standard is requested in `CMAKE_CXX_STANDARD`, which can still be set to any legal version".

So there is no problem for users to still select `-DCMAKE_CXX_STANDARD=17` or `-DCMAKE_CXX_STANDARD=20`.